src/poetry/console/commands/check.py

```diff
@@ -61,7 +59,8 @@ def handle(self) -> int:
         from poetry.factory import Factory
 
         # Load poetry config and display errors, if any
-        poetry_file = Factory.locate(Path.cwd())
+        project_path = self.poetry.file.path
+        poetry_file = Factory.locate(project_path)
         config = PyProjectTOML(poetry_file).poetry_config
```

Comment:
Can't we use `self.poetry.file.path` directly as input for `PyProjectTOML`? :thinking:

Reply:
Seems reasonable enough, project_path and poetry_file lead to the same path no matter if the file was specified by argument or determined implicitly. I'll test that change some more and push it up.